apps/web/src/app/api/oap_mcp/proxy-request.ts

```diff
@@ -89,6 +90,33 @@ async function getMcpAccessToken(supabaseToken: string, mcpServerUrl: URL) {
  * @returns The response from the MCP server.
  */
 export async function proxyRequest(req: NextRequest): Promise<Response> {
+  // Extract the path after '/api/oap_mcp/'
+  // Example: /api/oap_mcp/foo/bar -> /foo/bar
+  const url = new URL(req.url);
+  const path = url.pathname.replace(/^\/api\/oap_mcp/, "");
+
+  // Check if the first path segment might be a server name
+  const pathSegments = path.split("/").filter(Boolean);
+  if (pathSegments.length > 0) {
+    const servers = getMCPServers();
+    const potentialServerName = pathSegments[0];
+
+    // If the first segment matches a configured server, delegate to new proxy
+    if (servers[potentialServerName]) {
+      // Import and use the new per-server proxy handler
+      const { proxyRequest: newProxyRequest } = await import(
+        "./[server]/[...path]/route"
+      );
+      return newProxyRequest(req, {
+        params: {
+          server: potentialServerName,
+          path: pathSegments.slice(1),
+        },
+      });
+    }
```

Comment:
This is throwing an error for me.

Switching to dynamic import with const { POST: newProxyRequest } = await import("./[server]/route") and calling the handler with params: { server: potentialServerName } works correctly for per-server proxying in Next.js route handlers. The try/catch ensures graceful fallback to legacy logic if the import fails. This approach resolves import and routing issues present in the previous implementation.